src/R/Components/Impl/InteractiveWorkflow/Commands/SessionInformationCommand.cs

```diff
@@ -27,9 +28,39 @@ public sealed class SessionInformationCommand : IAsyncCommand {
             }
         }
 
-        public Task<CommandResult> InvokeAsync() {
-            _interactiveWorkflow.RSessions.PrintBrokerInformation();
-            return Task.FromResult(CommandResult.Executed);
+        public async Task<CommandResult> InvokeAsync() {
+            await PrintBrokerInformationAsync();
+            return CommandResult.Executed;
+        }
+
+        private void OnBrokerChanged(object sender, EventArgs e) {
+            if (_interactiveWorkflow.RSession.IsRemote) {
+                PrintBrokerInformationAsync(reportTelemetry: true).DoNotWait();
+            }
+        }
+
+        private async Task PrintBrokerInformationAsync(bool reportTelemetry = false) {
+            var a = await _interactiveWorkflow.RSessions.Broker.GetHostInformationAsync();
+            var console = _interactiveWorkflow.RSessions.Console;
```

Comment:
On the level of `InteractiveWorkflow`, we can write directly to the interactive window - no need to reference it back from `RSession`